packages/nimble-components/src/utilities/models/update-tracker.ts

```diff
@@ -0,0 +1,26 @@
+import { Tracker } from './tracker';
+
+/**
+ * Generic Update Tracker Utility which extends Tracker Utility with update logic
+ */
+export abstract class UpdateTracker<
+    TrackedProperties extends readonly string[]
+> extends Tracker<TrackedProperties> {
+    public constructor(requiredUpdates: TrackedProperties) {
+        super(requiredUpdates);
+    }
+
+    public trackAndQueue(
+        key: TrackedProperties extends readonly (infer U)[] ? U : never
+    ): void {
+        this.track(key);
+        this.queueUpdate();
+    }
+
+    public trackAllAndQueue(): void {
+        this.trackAll();
+        this.queueUpdate();
+    }
```

Comment:
I don't think these are adding significant value. Instead of developers just having to understand when queue calls happen they have to understand when track or trackAndQueue are called. Also trackAllAndQueue seems completely unused?
I think you can avoid growing the api surface in these cases